lib/basis_transcode.cpp

```diff
@@ -358,10 +358,14 @@ ktxTexture2_transcodeUastc(ktxTexture2* This,
         memcpy(priv._levelIndex, protoPriv._levelIndex,
                This->numLevels * sizeof(ktxLevelIndexEntry));
         // Move the DFD and data from the prototype to This.
-        delete This->pDfd;
+        if (This->pDfd) {
```

Comment:
This `if` is not necessary for 2 reasons:

1. Previous checks would have prevented the function getting this far if the texture lacked a DFD pointer.
2. The `free` man page states "If ptr is a NULL pointer, no operation is performed.".

Please remove it.

Reply:
Sure, no problem. I saw it at few places in the code, I assumed it was the convention, texture.c and texture2.c in particular.